### ISSUE TITLE
Bonsai | UI Feature request:  promote Grouping and Filtering to main menu

### ISSUE THREAD

**Issue by steverugi** [IfcOpenShell/IfcOpenShell#5857]
As previously mentioned on OSArch, it would be extremely convenient to have the `Grouping and Filtering` panel accessible directly from the main menu. Something like:

![image](https://github.com/user-attachments/assets/319f9823-80b0-4ae1-9e4a-b4da5e0e8a82)

In my opinion it is an essential feature, currently available at the very bottom, requiring frequent scrolling to access.

I hope this meets the preferences of both users and developers.
Thank you in advance

**Comment by sboddy** [comment]
https://community.osarch.org/discussion/2649/bonsai-ui-feature-request-promote-grouping-and-filtering-to-main-menu#latest

Just linking some forum feedback on this proposal.

**Comment by Netrieb** [comment]
I would even suggest to place it in to the `Outliner area` because Filtering, Searching and the Outliner are very similar things, wouldn't it make more sense? When you do your search, you have immediate visual feedback in the Outliner, you could have the Search tab still visible etc.

![Image](https://github.com/user-attachments/assets/926df695-fa53-49f7-8436-5d575ab7fcd8)

**Comment by Moult** [comment]
Although it is possible for us to inject a new button at the top of the
outliner which will trigger a popup panel or similar to show grouping
and filtering there, it's probably not a good idea because it reduces
trust in the user on "where to look for features".

In the past we had lots of features spread about everywhere in the
Blender UI and so it wasn't clear where a feature might be. I'd say a
lot of users never discovered some of the features we had hidden in the
material node editor. Since we introduced our own tabbed UI, that made
it clear that everything in these tabs is everything we have to offer.

I think it makes sense to put these at hte top of the Quality and
Coordination tab. I've assigned this task to @blenderdefender

**Comment by sboddy** [comment]
Quality and Coordination? 🤔 I still consider myself a newbie, but that strikes me as an illogical place to put Grouping and Search. I thought my suggestion in the linked discussion of moving it up in the Project Overview tab _above_ the "set and forget" stuff was an OK compromise.

**Comment by Moult** [comment]
I think you're right, so just below the Spatial panel, still in Project information?

**Comment by steverugi** [comment]
glad to see some action about my proposal ;)
the initial idea for me was not to have a popup window but rather a dedicated panel among the existing ones

Totally agree with @sboddy , Quality and Coordination doesn't seem the _most obvious drawer where to find your socks_

I am saying this as a user who daily goes through the menu countless times and features in Grouping and Filtering are the mostly used, it would be interesting to know what other _daily users_  think about it.

As much as I love Bonsai, and its community, the panels on the right are still a bit of a challenge even though developing muscle memory helps navigating through them, I particularly like the recently updated `T` nested icon menu for instance, it's far more intuitive and convenient.

Maybe in the future a different approach could be considered/developed? I find the "tiles" used in Inkscape or QGIS a handy option, it allows the selection and arrangement of the panels you need without clogging screen space.

In any case, as long as Grouping and Filtering is positioned above and (equally important) instance grouping management comes out of the hidden "misc" I'm happy, it's a very useful step forward

I truly appreciate it, thank you

**Comment by Netrieb** [comment]
@steverugi In Blender You can drag the tools: 

https://github.com/user-attachments/assets/d9c30878-bc1d-432a-b4a3-92cb142de5dc

**Comment by sboddy** [comment]
@Netrieb 
> In Blender You can drag the tools:

Yes, but you would have to repeat the drag and drop every time you restart Blender. Bonsai panel ordering is not "sticky".

**Comment by Netrieb** [comment]
> [@Netrieb](https://github.com/Netrieb)
> 
> > In Blender You can drag the tools:
> 
> Yes, but you would have to repeat the drag and drop every time you restart Blender. Bonsai panel ordering is not "sticky".

You can make it permanent: `File -> Defaults -> Save startup file`

**Comment by sboddy** [comment]
> You can make it permanent: `File -> Defaults -> Save startup file`

Unfortunately the only way to do this is to create a project so that I can move the panel. Then when I restart I have the remnants of that unwanted project polluting the tree, but not active as a project. All the Bonsai panels are now in pre-project mode, so I have to create _another_ project, doubling all the structure and empties under Scene Collection. So... I have to force delete all those entries, save as the default file _again_ and restart Bonsai again. _Now_ Bonsai keeps the Grouping and Filtering panel where I wanted it, and I can create a new project without a polluted structure. OK, for me to do, I'm not sure we want to ask everyone to do that fun little dance.

[Edit: Don't get me wrong, it is a valid workaround, just a bit ugly for newbies who are trying to learn Bonsai/Blender.]

**Comment by sboddy** [comment]
> > You can make it permanent: `File -> Defaults -> Save startup file`
> 
> Unfortunately the only way to do this is to create a project so that I can move the panel. Then when I restart I have the remnants of that unwanted project polluting the tree, but not active as a project. All the Bonsai panels are now in pre-project mode, so I have to create _another_ project, doubling all the structure and empties under Scene Collection. So... I have to force delete all those entries, save as the default file _again_ and restart Bonsai again. _Now_ Bonsai keeps the Grouping and Filtering panel where I wanted it, and I can create a new project without a polluted structure. OK, for me to do, I'm not sure we want to ask everyone to do that fun little dance.

[Edit: Don't get me wrong, it _is_ a valid workaround, just a bit ugly for newbies who are trying to learn Bonsai/Blender.]

**Comment by steverugi** [comment]
yep I had the same experience any time I tried to customize Bonsai's UI by the 'save startup file'-thing
@sboddy 's solution is smart but I think unlikely something to be advised as regular procedure to anyone, regardless of their skills

I understand getting away from Outliner is not an easy task, my personal opinion is that, to do so, we need something to complement Spatial Decomposition to perform a smoother and intuitive workflow

I was thinking:
 
Parts of Project Setup, Geometry, and Stakeholders' panels could be in a popup window activated by an icon somewhere, especially those fature used for something you only do once in a while or just at the beginning, or maybe never if you use custom template .ifc file, they _cumbersomely_ sit in a place where you'd need tools used more frequently, as a DIY guy that's the way I see it. 

Conversely, and in line with @Netrieb above, Spatial Decomposition, Grouping and Filtering, Status Filter (never understood why it ended up under Costing and Scheduling where I would aptly see Spredsheet Import/Export) could be those to dethrone (and get rid of) Outliner, gaining their own panel on the left.

just my 2 cents on the matter, mind you, thank you for your ideas, always a pleasure

**Comment by Moult** [comment]
FYI the status filter is there because it is the basic form of time based data (i.e. scheduling).

**Comment by steverugi** [comment]
> FYI the status filter is there because it is the basic form of time based data (i.e. scheduling).

I thought that status selection/view was something usable in a broader context, not specifically applicable to scheduling per se

thanks for the explanation

**Comment by OrionSehn** [comment]
# Summary of discussion, and proposed next steps

I'm gonna try and summarize and refocus the discussion above for those just looking at this now. 

## Problem: Grouping and Filtering Feature is Not Easily Findable

Its hard for users to find the grouping and filtering feature in Bonsai right now. Its a really key feature that should be immediately available to users as its one of the most common actions you need to take when working on a BIM file. 

Original proposal was to make it a new tab in the Bonsai Bar, however this may set a precedent for UI/UX scope creep.

![Image](https://github.com/user-attachments/assets/42b51ba9-1c10-4f73-9a51-3c04db55bd9d)

There was some other proposals, ie. making a new popup window, adding it to the Quality and Coordination tab. It seems there's some degree of agreement on keeping it in the project overview tab, but moving it up. 

## Proposal: Keep Grouping and Filtering in the Project Overview tab, but move it up.

### Details
- This way users won't have to scroll to the bottom to find the Grouping and Filtering functionality
- Because the Project Overview page is the default page, it'll be made immediately available to users, so this should increase the visibility of this function significantly.
- I propose that we redirect the tooltip link to documentation on grouping and filtering to the following link:
https://docs.bonsaibim.org/reference/project_overview/index.html#grouping-and-filtering 
  - List out definitions of between class/attribute/property/etc.
  - Include some screenshots of using the grouping/filtering tool for filtering in some simple examples. 
  - Show one example of the selector syntax and link to more detailed documentation https://docs.ifcopenshell.org/ifcopenshell-python/selector_syntax.html


### Specific Steps: 

- [ ] Move Grouping/Filtering Functionality panel up
- Extend the Project Overview Documentation
  - [ ] Definitions of class/attribute/property/etc
  - [ ] Simple Selecting/Filtering Examples
  - [ ] 1 selector syntax example
  - [ ] Link more complex, selector syntax for more sophisticated queries

@Moult Please approve/provide suggestions to the above changes when you can spare an ear

**Comment by talukderpushpak2000** [comment]
Hi everyone, I am a new developer and I am really interested in contributing to this issue. I would love to give this issue a try and will work on making a pull request to improve the accessibility of the "Grouping and Filtering" panel. I understand how important this feature is for usability and I am open to any suggestions or guidance from the maintainers.

**Comment by steverugi** [comment]
@OrionSehn 
thanks for your post, I struggle to understand the meaning of your 
> Original proposal was to make it a new tab in the Bonsai Bar, however this may set a precedent for UI/UX scope creep.

Genuine question: is not "Search" always on top of any UI/UX everywhere? from Windows to Google to any website including OSArch? 
The current arrangement in Bonsai is rather unintuitive, as a staunch supporter of Bonsai I don't say this too lightly

Someone proposed a turn on/off option of the Scene tabs, something I appreciate with Inkscape or Qgis, fully customizable and convenient with smaller screens, when you have many options (like in Bonsai) it's a good feature.

I think it's time to kick out the 'Outliner' panel as default and replace it with something more productive, true that panels can be rearranged but there is no chance, to the best of my knowledge, to make it permanent.

cheers

**Comment by steverugi** [comment]
Hi @talukderpushpak2000 

> I would love to give this issue a try and will work on making a pull request to improve the accessibility of the "Grouping and Filtering" panel. I understand how important this feature is for usability and I am open to any suggestions or guidance from the maintainers.

Thanks for your input
If interested, my personal opinion is to 'prune' the current panels of non-recurrent ones, like 'Project Setup' for instance.
Maybe it's too drastic a solution to send them to the Extension Preferences window but something similar would free some space for panels that are regularly used.

And I'd like Spreadsheet Import/Export moved to Costing and Scheduling, not for myself (I'm used to it) but in view of providing easy access to beginners, in my perspective it makes much more sense.

cheers

**Comment by OrionSehn** [comment]
> [@OrionSehn](https://github.com/OrionSehn) thanks for your post, I struggle to understand the meaning of your
> 
> > Original proposal was to make it a new tab in the Bonsai Bar, however this may set a precedent for UI/UX scope creep.
> 
> Genuine question: is not "Search" always on top of any UI/UX everywhere? from Windows to Google to any website including OSArch? The current arrangement in Bonsai is rather unintuitive, as a staunch supporter of Bonsai I don't say this too lightly
> 
> Someone proposed a turn on/off option of the Scene tabs, something I appreciate with Inkscape or Qgis, fully customizable and convenient with smaller screens, when you have many options (like in Bonsai) it's a good feature.
> 
> I think it's time to kick out the 'Outliner' panel as default and replace it with something more productive, true that panels can be rearranged but there is no chance, to the best of my knowledge, to make it permanent.
> 
> cheers

I was summarizing some of the discussion, for those who found the whole thread, and associated forum posts too much to read. The opinion of scope creep comes from this thread: https://community.osarch.org/discussion/2649/bonsai-ui-feature-request-promote-grouping-and-filtering-to-main-menu#latest 

My personal opinion on the matter, is that I agree that searching and filtering is so fundamental, that it should be a top level feature. I'd personally love to see a SQL style interface where you put in the IFCOpenShell Selector Syntax, and press a run button to select the elements that are relevant to you. 

Though I'm sure there's business pressure to support and emphasize the other major, more ambitious features like Drawings and Documents, Services and Systems, and Facility Management. But for me, searching and filtering is necessary for working with a model and should be front and center.

@talukderpushpak2000 maybe after you make your first pull request, where you just try rearranging existing panels, you can try and build out a second tab, and see if its worth exploring further.

**Comment by Moult** [comment]
Shifted
